lxd/resources/system.go

```diff
@@ -0,0 +1,328 @@
+package resources
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+)
+
+var sysClassDMIID = "/sys/class/dmi/id"
+var systemType string
+
+// GetSystem returns a filled api.ResourcesSystem struct ready for use by LXD
+func GetSystem() (*api.ResourcesSystem, error) {
+	var err error
+	system := api.ResourcesSystem{}
+
+	// Cache the system type
+	if systemType == "" {
+		// According to the man page, if a virtualization technology is detected, 0 is returned, a non-zero code otherwise.
+		out, err := shared.RunCommand("systemd-detect-virt")
+		if err != nil {
+			systemType = "physical"
+		} else {
+			systemType = out
+		}
```

Comment:
So I don't think we want to show `lxc` but just `physical`, `container` or `virtual-machine`.

Should we just use `systemd-detect-virt --container` and `systemd-detect-virt --vm`, if both return non-zero, then go physical?

Reply:
> Should we just use systemd-detect-virt --container and systemd-detect-virt --vm, if both return non-zero, then go physical?

Yes, we should do that. I thought `systemd-detect-virt` would only output `VM` or `Container` but apparently I was wrong ;)